tests/functional/CommonDBTM.php

```diff
@@ -1367,4 +1367,53 @@ function () use ($computer, $value) {
 
         $this->string($computer->fields['name'])->isEqualTo($truncated);
     }
+
+    public function testCheckUnicity()
+    {
+        $this->login();
+
+        $field_unicity = new \FieldUnicity();
+        $this->integer($field_unicity->add([
+            'name' => 'uuid uniqueness',
+            'itemtype' => 'Computer',
+            '_fields' => ['uuid'],
+            'is_active' => 1,
+            'action_refuse' => 1,
+            'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+        ]))->isGreaterThan(0);
+
+        $computer = new \Computer();
+        $this->integer($computers_id1 = $computer->add([
+            'name' => __FUNCTION__ . '01',
+            'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+            'uuid' => '76873749-0813-482f-ac20-eb7102ed3367'
+        ]))->isGreaterThan(0);
+
+        $this->integer($computers_id2 = $computer->add([
+            'name' => __FUNCTION__ . '02',
+            'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+            'uuid' => '81fb7b20-a404-4d1e-aafa-4255b7614eae'
+        ]))->isGreaterThan(0);
+
+        $this->variable($computer->update([
+            'id' => $computers_id2,
+            'uuid' => '76873749-0813-482f-ac20-eb7102ed3367'
+        ]))->isNotTrue();
+
+        // Expect 1 error in session messages
+        $this->array($_SESSION['MESSAGE_AFTER_REDIRECT'][1])->size->isEqualTo(1);
```

Comment:
There are dedicated methods in GLPITestCase class to check for (un)expected session messages

Reply:
`$this->hasSessionMessages()` method